File: Backend/TranscomaAPI/Comun/Entidades/Cliente.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TranscomaAPI.Comun.Entidades
{
    public class Cliente : Entidad
    {
        private string _nombre;
        private string _nombreUsuario;
        private string _correo;
        private string _password;
        private DateTime _fechaRegistro;
        private Administrador _administrador;

        /// <summary>
        /// 
        /// </summary>
        /// <param name="id"></param>
        /// <param name="nombre"></param>
        /// <param name="nombreUsuario"></param>
        /// <param name="correo"></param>
        /// <param name="password"></param>
        /// <param name="fechaRegistro"></param>
        public Cliente(int id, string nombre, string nombreUsuario, string correo, string password, DateTime fechaRegistro)
        {
            Id = Id;
            _nombre = nombre;
            _nombreUsuario = nombreUsuario;
            _correo = correo;
            _password = password;
            _fechaRegistro = fechaRegistro;
        }

        /// <summary>
        /// Contructor para la consulta de los clientes de un admin
        /// </summary>
        /// <param name="id"></param>
        /// <param name="nombre"></param>
        /// <param name="nombreUsuario"></param>
        /// <param name="correo"></param>
        /// <param name="fechaRegistro"></param>
        /// <param name="administrador"></param>
        public Cliente(int id, string nombre,  string nombreUsuario, string correo, DateTime fechaRegistro, Administrador administrador)
        {
            Id = Id;
            _nombre = nombre;      
            _correo = correo;
            _nombreUsuario = nombreUsuario;           
            _fechaRegistro = fechaRegistro;
            _administrador = administrador;
        }

        /// <summary>
        /// Contructor para las entradas, salidas y pedidos
        /// </summary>
        /// <param name="id"></param>
        /// <param name="nombre"></param>
        /// <param name="correo"></param>

        public Cliente(int id, string nombre, string correo)
        {
            Id = Id;
            _nombre = nombre;
            _correo = correo;
        }

        /// <summary>
        /// Getters y Setters
        /// </summary>        
        public string Nombre { get => _nombre; set => _nombre = value; }
        public string Login { get => _nombreUsuario; set => _nombreUsuario = value; }
        public string Correo { get => _correo; set => _correo = value; }
        public string Contraseña { get => _password; set => _password = value; }
        public DateTime FechaRegistro { get => _fechaRegistro; set => _fechaRegistro = value; }
        public Administrador Administrador { get => _administrador; set => _administrador = value; }
    }
}

File: frontend/src/app/models/Cliente.ts
import { EntidadBase } from "./EntidadBase";
import { Administrador } from "./Administrador";

export class Cliente extends EntidadBase {

    private _nombre: string;
    private _nombreUsuario: string;
    private _correo: string;
    private _password: string;
    private _fechaRegistro: string;
   // private _administrador: Administrador;

    constructor(data?: any) {
        super();
        if (data !== undefined) {
            this.$id = data.id;
            this.$nombre = data.nombre;
            this.$nombreUsuario = data.nombreUsuario;
            this.$correo = data.correo;
            this.$password = data.password;
           /* if(data.administrador !== undefined){
                this.$administrador =  new Administrador(data.administrador);
            }
            else{
                this.$administrador = null;
            }
            */
            this.$fechaRegistro = data.fechaRegistro;
            
        }
    }

    /** Getters y setters */
    public get $nombre(): string {
        return this._nombre;
    }

    public set $nombre(value: string) {
        this._nombre = value;
    }

    public get $nombreUsuario(): string {
        return this._nombreUsuario;
    }

    public set $nombreUsuario(value: string) {
        this._nombreUsuario = value;
    }

    public get $correo(): string {
        return this._correo;
    }

    public set $correo(value: string) {
        this._correo = value;
    }

    public get $password(): string {
        return this._password;
    }

    public set $password(value: string) {
        this._password = value;
    }

    public get $fechaRegistro(): string {
        return this._fechaRegistro;
    }

    public set $fechaRegistro(value: string) {
        this._fechaRegistro = value;
    }

  /*  public get $administrador(): Administrador {
        return this._administrador;
    }

    public set $administrador(value: Administrador) {
        this._administrador = value;
    }*/
}